### PR TITLE
Add SUMA integration ADR

### PR DESCRIPTION
**Pull request by nelsonkopliku**
As per the title: SUMA integration ADR.

### CODE REVIEW

File: adr/0013-suma-integration.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+Date: 2024-05-07
+
+## Status
+
+Accepted
+
+## Context
+
+Keeping SAP infrastructure up-to-date and secure is a crucial task for system administrators, allowing vulnerabilities reduction and overall system improvement in terms of features and performance.
```
Comment (arbulu89) [inline adr/0013-suma-integration.md:11]: To be honest, this initial chapter looks a bit "actually" out of context.
As it doesn't talk about any technical challenge.
It talks about about a product new feature.

I think the context should talk about the challenge of communicating with a 3rd party component that doesn't have any reactiveness, and we are forced to poll data in periodical basis.

File: adr/0013-suma-integration.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+These premises make it both **strategically and technically convenient** leveraging SUMA's capabilities within Trento to provide a comprehensive SUSE driven experience to SAP userbase.
+
+> Current integration only focuses on enhancing visibility by running software updates discoveries. No corrective actions are taken on the hosts by Trento.
+>
+> SUMA version, and related HTTP APIs, currently targeter for integration is [4.3](https://documentation.suse.com/suma/4.3/api/suse-manager/index.html)
+
+## Considered Options
+
+- **Eventless Integration**: just proxying SUMA API calls directly within Trento's APIs, without the need of interacting with Trento's event driven nature
```
Comment (arbulu89) [inline adr/0013-suma-integration.md:27]: I think that here, on top of the table, we had 2 real scenario, as the health usage on the host health was initially taken as granted, as it was a business requirement.

1. Query and show data coming directly from SUMA (with the disadvantage that the data coming might not be consistent with the host health)
2. Query and store data coming from SUMA, so the health of the host is calculated always with the latest discovered suma info

With that being said, I would said that this initial implementation goes with the 2nd option, as we don't have any capability by now to know if there are changes in SUMA without polling directly.
With this, I would say that the 1st option could be "workarounded" just dispatching the health updated command everytime we query SUMA, but we discarded for not adding this side effect in "read" api query

**Review by arbulu89 — COMMENTED** [review]
Hey @nelsonkopliku ,
I didn't have yet the chance to go through all of it.
I think that we should focus the ADR in the debate of "all proxied" or "proxied data save in trento DB".
Which was the part that has real impact on the decision and the real dilema.
The fact of having the health affected is a business decision so not really something we could really architect.

In my opinion, we should comment 2 options, and why we chose the 2nd by now.
We can expose as well that we do a scheduled query as SUMA doesn't have any way of notifying 3rd parties of its changes.

I think with this we would cover all the main discussions we had.
For me, personally, everything else is not really a architectural debate, because there weren't real dilemas. They were things that actually came from our Product owner in his tests (for example, that the a discovery must be made on host registered). We could list them if needed in a simple bullet list fashion, without going to complex.

That being said, I want to hear the opinion for others as well. Maybe having such a super detailed record is something they like

**Comment by nelsonkopliku** [comment]
Thanks for the feedback @arbulu89!

So, yes, there may be different focus levels.

I'd be fine in re-scoping only on _track vs non-track suma results_.

I more or less unconsciously went with this exhaustiveness to give comprehensive overview to newcomers :smile:

**Comment by nelsonkopliku** [comment]
Hey @arbulu89, I still think there was interesting information in the initial _extended version_, however I see that maybe not for this specific ADR. Rescoped to what suggested :+1:

**Review by rtorrero — APPROVED** [review]
LGTM

**Review by arbulu89 — APPROVED** [review]
@nelsonkopliku 
I like much much more this version, actually I find it really good.
And I appreciate the effort you put in the other condensed version, but I personally think that it was a bit out of context for a ADR.
Maybe we could put that other content in some other `doc`.

Thank you man!!

**Comment by nelsonkopliku** [comment]
Thanks @arbulu89!

> Maybe we could put that other content in some other doc.
I was actually thinking about that, so I kept a copy of that version :smile:

**Comment by dottorblaster** [comment]
Sorry, late to the party.

I also wanted to praise the new version of the ADR, thank you for this @nelsonkopliku :+1:

**Pull request merged**